### PR TITLE
Bugfix: erroneous use of cython wrap around

### PR DESCRIPTION
**Pull request by hmaarrfk**
I don't understand why this wasn't throwing errors.

There may only be a few places where this is needed. depending on the conclusion from this clarification request: https://github.com/cython/cython/issues/2667

@jni @AetherUnbound

## Checklist
[It's fine to submit PRs which are a work in progress! But before they are merged, all PRs should provide:]
- [ ] Unit tests??? Man this memory stuff is really hard to test. Not like I can create explicit failers easily

## For reviewers
- [ ] Check that the PR title is short, concise, and will make sense 1 year
  later.
- [ ] Check that new functions are imported in corresponding `__init__.py`.
- [ ] Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.
- [ ] Consider backporting the PR with `@meeseeksdev backport to v0.14.x`

### CODE REVIEW

**Comment by stefanv** [comment]
It is interesting that Cython never complained about this; any idea why?!

**Comment by stefanv** [comment]
(Approving, even though it is a WIP, since I think any changes in line with these should go in.)

File: skimage/restoration/_denoise_cy.pyx [hunk 1]
```diff
@@ -193,13 +193,14 @@ def _denoise_tv_bregman(image, double weight, int max_iter, double eps,
         double rmse = DBL_MAX
         double norm = (weight + 4 * lam)
 
-    u[1:-1, 1:-1] = image
+    height, widht = image.shape[:2]
```
Comment (jni) [inline skimage/restoration/_denoise_cy.pyx]:
```suggestion
    height, width = image.shape[:2]
```

File: skimage/segmentation/_felzenszwalb_cy.pyx [hunk 2]
```diff
@@ -52,28 +52,28 @@ def _felzenszwalb_cython(image, double scale=1, sigma=0.8,
...
-    down_cost = np.sqrt(np.sum((image[1:, :, :] - image[:-1, :, :])
-    	*(image[1:, :, :] - image[:-1, :, :]), axis=-1))
-    right_cost = np.sqrt(np.sum((image[:, 1:, :] - image[:, :-1, :])
-    	*(image[:, 1:, :] - image[:, :-1, :]), axis=-1))
-    dright_cost = np.sqrt(np.sum((image[1:, 1:, :] - image[:-1, :-1, :])
-	*(image[1:, 1:, :] - image[:-1, :-1, :]), axis=-1))
-    uright_cost = np.sqrt(np.sum((image[1:, :-1, :] - image[:-1, 1:, :])
-    	*(image[1:, :-1, :] - image[:-1, 1:, :]), axis=-1))
+    down_cost = np.sqrt(np.sum((image[1:, :, :] - image[:height-1, :, :]) *
+    	                         (image[1:, :, :] - image[:height-1, :, :]), axis=-1))
```
Comment (hmaarrfk) [inline skimage/segmentation/_felzenszwalb_cy.pyx]: @jni, this kind of indentation makes it clear that the goal was to compute the Frobenius norm. I really wish numpy had this kind of optimization. Maybe this is where pythran comes in!

I'll leave that optimization to an other PR that actually cares to do a benchmark for this.

Comment (jni) [reply skimage/segmentation/_felzenszwalb_cy.pyx]: @hmaarrfk if you really want to see optimization, try using `@` here! ;) Although the discontiguity might break it.

Comment (hmaarrfk) [reply skimage/segmentation/_felzenszwalb_cy.pyx]: Na, I listened to @serge-sans-paille's talks, and pythran might be the future

**Comment by hmaarrfk** [comment]
@jni I think this is ready.

**Pull request merged**